src/services/services.ts

```diff
@@ -5762,6 +5773,62 @@ namespace ts {
             return forEach(diagnostics, diagnostic => diagnostic.category === DiagnosticCategory.Error);
         }
 
+        function getDependencies(fileName: string): DependencyInfo {
+            synchronizeHostData();
+            
+            let sourceFile = getValidSourceFile(fileName);
+            if (!sourceFile) {
+                return undefined;
+            }
+            
+            let resolver = program.getDiagnosticsProducingTypeChecker().getEmitResolver(sourceFile);
+            
+            let runtime: string[] = [];  
+            let compileTime: string[] = [];  
+            for (let node of sourceFile.amdDependencies) {  
+                runtime.push(node.path);  

```

Comment:
honoring these is dependent on the module target, if it is AMD or UMD, we use them, otherwise they are ignored i.e. in SystemJS and CommonJS. it might be useful to split them in their own list, instead of in "runtime". this way you can have: "imports", "amdDependencies", and "typeOnlyImports"/"designTimeImports"